viewing_party/main.py

```diff
@@ -0,0 +1,129 @@
+movie_dict = {} 
+user_data = {}
+#user_data = {"watched":[{movie},{},{}...],
+#              "watchlist":[{movie},{},{}...],
+#              "friends":[{friend},{},{}...],
+#              "subscriptions": ["string","",""]
+#              "favorites": [{movie},{},{}...]}
+#{movie} --> {"title":"string" , "genre":"string" , "rating":float}
+#{friend} --> {"watched": [{movie2},{},{}...]}
+#{movie2} --> {"title":"string" , "host": "streaming"}
+
+
+#Wave 1 
+#Q1 - add a new movie(title,genre,rating) to movie_dict
+def create_movie(title,genre,rating):
+    if not title or not genre or not rating:
+        return None
+    else:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict
+
+#Q2 - add a movie to list "watched"
+def add_to_watched(user_data,movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+#Q3 - add a movie to list "watchlist"
+def add_to_watchlist(user_data,movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+#Q4 - move a movie from "watchlist" to "watched" if in watched list, otherwisde return user_data
+def watch_movie(user_data,title):
+    watchlist = user_data["watchlist"]
+    watched = user_data["watched"]
+    for movie in watchlist:
+        if title in movie["title"]:
+            watchlist.remove(movie)
+            watched.append(movie)
+            return user_data
+    else:
+        return user_data
+
+
+#Wave 2: 
+#Q1 - Avg rating of watched
+def get_watched_avg_rating(user_data):
+    ratings = []
+    for movie in user_data["watched"]:
+        ratings.append(movie["rating"])
+    if len(ratings) == 0:
+        avg_ratings = 0 
+    else:    
+        avg_ratings = sum(ratings)/len(ratings)
+    return avg_ratings
+
+#Q2 - Most watched genere 
+def get_most_watched_genre(user_data):
+    if len(user_data["watched"]) == 0: 
+        return None
+    else:
+        genres = []
+        for movie in user_data["watched"]:
+            genres.append(movie["genre"])
+        genres_count = {}
+        for genre in genres:
+            if genre in genres_count:
+                genres_count[genre] +=1
+            else:
+                genres_count[genre] =1
+        fav_genre = max(genres_count, key=genres_count.get)
```

Comment:
This works, and is concise! It's also a good exercise to write a max function yourself where you loop through a check if the current value is higher than your current highest.